docs/admin/scale/index.md

```diff
@@ -0,0 +1,36 @@
+We regularly scale-test Coder against various reference architectures. Additionally, we provide a [scale testing utility](#scaletest-utility) which can be used in your own environment to give insight on how Coder scales with your deployment's specific templates, images, etc.
+
+## Reference Architectures
+
+| Environment                                       | Users         | Last tested  | Status   |
+| ------------------------------------------------- | ------------- | ------------ | -------- |
+| [Google Kubernetes Engine (GKE)](./gke.md)        | 50, 100, 1000 | Nov 29, 2022 | Complete |
+| [AWS Elastic Kubernetes Service (EKS)](./eks.md)  | 50, 100, 1000 | Nov 29, 2022 | Complete |
+| [Google Compute Engine + Docker](./gce-docker.md) | 15, 50        | Nov 29, 2022 | Complete |
+| [Google Compute Engine + VMs](./gce-vms.md)       | 1000          | Nov 29, 2022 | Complete |
+
+## Scale testing utility
+
+Since Coder's performance is highly dependent on the templates and workflows you support, we recommend using our scale testing utility against your own environments.
+
+For example, this command will do the following:
+
+- create 100 workspaces
+- establish a SSH connection to each workspace
+- run `sleep 3 && echo hello` on each workspace via the web terminal
+- close connections, attempt to delete all workspaces
+- return results (e.g. `99 succeeded, 1 failed to connect`)
```

Comment:
In scale testing, concurrency matters a whole lot.  What things are concurrently happening in our tests?

Are all the workspace creates happening concurrently?

Are any SSH connections being set up concurrently with creates?

Are all the SSH connections being set up concurrently?

Are the commands running concurrently with each other and with other steps?

etc.

Reply:
I believe the answer is "yes" to all of the above, but @deansheather can confirm.

I agree we should also have a sentence or two about this behavior in the CLI help and the docs. @deansheather could you help with that?

Reply:
The whole flow for one workspace is a single job (i.e. creating the user, creating the workspace, waiting for workspace, connections etc.), and the concurrency for jobs is controllable. Because the jobs run in "slices" all web terminal tests might not run simultaneously.

I'm thinking about having specific tests for just web terminal and connections which can be used against pre-existing workspaces

Reply:
> and the concurrency for jobs is controllable

Can you elaborate? I was looking at #5202 and am a bit confused about which flags control the concurrency. Stuff like `--no-wait-for-agents`?

Reply:
I actually forgot to specify the concurrency flags in the issue, my bad. We'll have flags for `--concurrency`, `--cleanup-concurrency`, `--timeout` and `--cleanup-timeout`

Reply:
@deansheather @spikecurtis I added the following line. Is this accurate?

https://github.com/coder/coder/blob/b31e8132f38e352360688b7ca9892223aa72e142/docs/admin/scale/index.md?plain=1#L38

Reply:
Sounds fine to me